### ISSUE TITLE
Hatch `docs ` env is broken.

### ISSUE THREAD

**Issue by waylan** [mkdocs/mkdocs#3517]
The changes in #3502 break the hatch env `docs` in that it no longer installs MkDocs in editable mode from the working tree. Instead it installs the latest PyPI package. Yes, mkdocs build/serve commands will build the documentation from the working tree, but the theme files are used from the PyPI package. it does not even build and install from the working tree, which would at least allow a quit and rerun to work. And that means there is to way to preview changes to templates or CSS files. Annoyingly, the file watcher is watching those files so we get a rebuild and refresh, but the watched/changed files are not actually used.

Consider the following abridged session:

```bash
$ hatch --env docs shell
(docs) $ python -m pip freeze
...
mergedeep==1.3.4
mkdocs==1.5.3
mkdocs-autorefs==0.5.0
...
(docs) $ python -m pip install -e .
...
(docs) $ python -m pip freeze
...
mergedeep==1.3.4
-e git+https://github.com/waylan/mkdocs.git@4a61b0eb130e450851d0704c10732c69a6b41c0d#egg=mkdocs
mkdocs-autorefs==0.5.0
...
```

Now at least I can run `mkdocs serve` from the Hatch shell. And yes, the env is retained upon exit and reentry so `hatch run docs:serve` also works, But once the env needs rebuilt, I will need to manually install mkdocs in editable mode again.

**Comment by waylan** [comment]
> Yes, mkdocs build/serve commands will build the documentation from the working tree,

Actually, I'm not sure that is happening either, but haven't tested it. I didn't confirm one way or the other before manually installing in editable mode and haven't reverted to check. However, it is peculiar that after doing an editable install I started seeing the following with each rebuild:

```
INFO    -  Building documentation...
INFO    -  Doc file 'dev-guide/translations.md' contains a link
           '../about/contributing.md#installing-for-development', but the doc
           'about/contributing.md' does not contain an anchor '#installing-for-development'.
```

Presumably that is due to a change made in #3511. But is didn't show up until after I installed in editable mode.  

In other words, this suggests that perhaps the `docs` dir in the working tree is also being ignored.

**Comment by oprypin** [comment]
Thanks. Both issues fixed now.
